### ISSUE TITLE
[Request] Publish to F-Droid

### ISSUE THREAD

**Issue by grravity** [deptofdefense/AndroidTacticalAssaultKit-CIV#57]
Highly suggesting submitting a build to F-Droid.
https://f-droid.org/en/docs/Submitting_to_F-Droid_Quick_Start_Guide/
https://f-droid.org/en/docs/Inclusion_Policy/
https://f-droid.org/

**Comment by kenktk** [comment]
I agree. I am on a FOSS only version of Android and downloading APKs through a website or downloading from Aurora store is a hassle. I would love to see it on F-Droid eventually.

**Comment by Kodmod** [comment]
Massive win if this gets on F-Droid

**Comment by nicopace** [comment]
Wanted to post here what I've posted on the discord server for reference, and also to seek advice or accompaniment on this endevour :) https://discord.com/channels/698067185515495436/725550034770198588/1286394532417638441

> 
> Hi folks,
> 
> I have been searching through the discord server to figure out if an fdroid submission has happened or not.
> It seems that there hasn't been any before.
> I would like this to happen.
> 
> So, I have some questions based on [f-droid inclusion policy](https://f-droid.org/es/docs/Inclusion_Policy/):
> 1. The original app author has been notified (and does not oppose the inclusion): do you know of any contact person to get this notification? I would just publish a soft-fork of it, but would be better if DoD would be ok with it
> 2. based on aurora store depencies, it doesn´t depend on play services
> 3. no proprietary tracking services, based on https://reports.exodus-privacy.eu.org/es/reports/com.atakmap.app.civ/latest/#trackers
> 4. no proprietary ad libraries
> 5. no non-free build tools
> 
> Additionally:
> - they maintain the source code on github, but their release cycle doesn't seem to be as frequent as the binary release.
> - no unsolicited executable binary downloads
> - I am not sure if DoD owns civ.map.atakmap.com domain
> 
> Could you help me sort out these things, so I can successfully submit the app to f-droid?
> 
> Would you be interested in sharing this exploration?
> 
> Thanks!
>

**Comment by nicopace** [comment]
FYI I sent this message to the discord server, and people have replied that I should email support@tak.gov
Not sure if this is the way to go, but reposting here in case there are better ways to approach this.
https://discord.com/channels/698067185515495436/725550034770198588/1286394532417638441

> Hi folks,
> 
> I have been searching through the discord server to figure out if an fdroid submission has happened or not.
> It seems that there hasn't been any before.
> I would like this to happen.
> 
> So, I have some questions based on [f-droid inclusion policy](https://f-droid.org/es/docs/Inclusion_Policy/):
> 1. The original app author has been notified (and does not oppose the inclusion): do you know of any contact person to get this notification? I would just publish a soft-fork of it, but would be better if DoD would be ok with it
> 2. based on aurora store depencies, it doesn´t depend on play services
> 3. no proprietary tracking services, based on https://reports.exodus-privacy.eu.org/es/reports/com.atakmap.app.civ/latest/#trackers
> 4. no proprietary ad libraries
> 5. no non-free build tools
> 
> Additionally:
> - they maintain the source code on github, but their release cycle doesn't seem to be as frequent as the binary release.
> - no unsolicited executable binary downloads
> - I am not sure if DoD owns civ.map.atakmap.com domain
> 
> Could you help me sort out these things, so I can successfully submit the app to f-droid?
> 
> Would you be interested in sharing this exploration?
> 
> Thanks!

**Comment by nicopace** [comment]
Some other relevant information from the [Building applications page](https://f-droid.org/es/docs/Building_Applications/):
- Although you could submit a binary APK, the ideal is for the source code to be buildable by fdroid
- It has to be signed by the same key as the other apk's , as no two apps with the same name can be installed (or update each other) with different signing keys, and uninstalling one app to install the other one may result in data loss

From the [Repository Style Guide](https://f-droid.org/es/docs/Repository_Style_Guide/):
- the F-Droid server needs a buildable source to build an APK from

From the [Inclusion How-To](https://f-droid.org/es/docs/Inclusion_How-To/):
- TBD

I will continue updating this post with the rest of the requirements.

I am happy to take this close to the finish line, but I won't be able to sign the apk... so would be good if this is something that is possible, or this approach doesn't make sense in the current context.

Thanks

**Comment by nicopace** [comment]
I just wanted to let people tracking this issue know that it is possible to compile the code using github actions, if you use a self-hosted runner.
You can check out my last successful run here: https://github.com/nicopace/AndroidTacticalAssaultKit-CIV/actions/runs/10981280460/job/30520849489

Had to do multiple attempts till I figured out that the build was failing because of not enough ram or storage.

This produces an installable APK.
